gcp_variant_transforms/libs/annotation/annotation_parser.py

```diff
@@ -0,0 +1,334 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for parsing annotation fields.
+
+The main class is `Parser`; see its documentation for usage. There are also
+some helper methods that can be used in different contexts.
+"""
+
+from __future__ import absolute_import
+
+import re
+
+from typing import Dict, List, Tuple  # pylint: disable=unused-import
+
+# The key in annotation maps that keeps the original alternate allele in an
+# annotation string (and the field name in the BigQuery table that holds that
+# information).
+ANNOTATION_ALT = 'allele'
+
+# The annotation field that VEP uses to record the index of the alternate
+# allele (i.e., ALT) that an annotation list is for.
+_ALLELE_NUM_ANNOTATION = 'ALLELE_NUM'
+
+# The representation of a deletion variant in VEP.
+_COMPLETELY_DELETED_ALT = '-'
+
+# Regular expressions to identify symbolic and breakend ALTs used in
+# annotation alt matching.
+# Check the VCF spec for symbolic and breakend ALT formats.
+_SYMBOLIC_ALT_RE = re.compile(r'^<(?P<ID>.*)>$')
+_BREAKEND_ALT_RE = (re.compile(
+    r'^(?P<up_to_chr>.*([\[\]]).*):(?P<pos>.*)([\[\]]).*$'))
+
+class AnnotationParserException(Exception):
```

Comment:
nit: styleguide requires two blank lines before every class definition. I know it would make this block longer, but I prefer to conform to the styleguide.

Reply:
Done.
Yes I knew I am breaking the style guide, in fact I even wanted `pass` to be on the same line but pylint complained :-)
I think this is a good/concise exception pattern but unfortunately the style guide rules make it too verbose (4 lines for each instead of one).